src/System.Runtime.Extensions/tests/Performance/Perf.ArrayList.cs

```diff
@@ -0,0 +1,454 @@
+            int elementsCollectionSize = (int)Benchmark.InnerIterationCount;
+            foreach (var iteration in Benchmark.Iterations)
+            {              
+                ArrayList[] elementsCollection = new ArrayList[elementsCollectionSize];
+                for (int index = 0; index < elementsCollectionSize; index++)
+                {
+                    elementsCollection[index] = new ArrayList(elements);
+                }                    
+                
+                using (iteration.StartMeasurement())
+                {
+                    for (int index = 0; index < Benchmark.InnerIterationCount; index++)
+                    {
+                        elementsCollection[index].Clear();
+                    }                        
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 1000)]
+        public void Contains()
+        {
+            int size = 100000;
+            ArrayList elements = CreateArrayListOfInts(size);
+            object contained = elements[size / 2];
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        elements.Contains(contained);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 100000000)]
+        public void Ctor_Default()
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            { 
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        new ArrayList();
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 10000)]
+        public void Ctor_ICollection()
+        {
+            ArrayList elements = CreateArrayListOfInts(10000);
+            object[] array = elements.ToArray();
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        new ArrayList(array);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 100000000)]
+        public void Count()
+        {
+            ArrayList elements = CreateArrayListOfInts(10000);
+            int temp = 0;
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        temp = elements.Count;
```

Comment:
If it can't already, I think it's possible at some point in the future the JIT will be able to devirtualize the Count call, at which point the JIT might eliminate this whole thing.  I'd suggest changing `temp` to instead be a static volatile field.

Reply:
Hope I fixed this correctly...